docs/cli/development.md

```diff
@@ -0,0 +1,234 @@
+# CLI Development Guide
+
+This guide provides detailed information for contributing to the Nixopus CLI development.
+
+## Project Structure
+
+```
+cli/
+├── main.py              # Main CLI entry point
+├── pyproject.toml       # Project configuration
+├── commands/            # Command implementations
+│   ├── version/         # Version command module
+│   ├── test/            # Test command module
+│   └── preflight/       # Preflight command module
+├── core/                # Core functionality
+│   ├── config.py        # Configuration utilities
+│   ├── version/         # Version display logic
+│   └── test/            # Test functionality
+├── utils/               # Utility functions
+└── tests/               # Test files
+```
+
+## Development Setup
+
+1. **Clone and Install**
+   ```bash
+   git clone https://github.com/your-org/nixopus.git
+   cd nixopus/cli
+   pip install -e ".[dev]"
```

Comment:
If we are using poetry for managing dependencies, lets stick to poetry for setup and avoid pip?